### PR TITLE
Try a python 3.10 CI job

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

**Comment by Cadair** [comment]
Upstream is unblocked.

**Comment by nabobalis** [comment]
It is out on pypi.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -62,6 +62,7 @@ stages:
 
         envs:
           - linux: py39
```
Comment (nabobalis) [inline azure-pipelines.yml:64]: I think it would be worth removing this and changing the online build to be 39

File: azure-pipelines.yml [hunk 2]
```diff
@@ -159,8 +159,8 @@ stages:
           test_command: 'pytest -p no:warnings --doctest-rst -m "not mpl_image_compare" --pyargs sunpy'
           submodules: false
           targets:
-            - wheels_cp3[789]-manylinux*x86_64
-            - wheels_cp3[789]-macosx*
+            - wheels_cp3[78910]-manylinux*x86_64
+            - wheels_cp3[78910]-macosx*
```
Comment (nabobalis) [inline azure-pipelines.yml:163]: This just seems very wrong.

Comment (Cadair) [reply azure-pipelines.yml:163]: yeah this doesn't work, this regex only matches one character.

When I did Astropy recently I couldn't find a functioning regex so I split them into their own jobs. The other option is use the env vars that cibuildwheel provides and then just match all Pythons here.

**Comment by dstansby** [comment]
I've opened https://github.com/asdf-format/asdf/issues/1027 for the `asdf` issue we're seeing

File: azure-pipelines.yml (hunk 2 above)
Comment (dstansby) [reply azure-pipelines.yml:163]: This is the real millenium bug.

Comment (nabobalis) [reply azure-pipelines.yml:163]: Separate targets it is.

**Comment by nabobalis** [comment]
Fails are asdf:

```python
../../.tox/py310/lib/python3.10/site-packages/asdf/asdf.py:174: in __init__
    self.tree = tree
../../.tox/py310/lib/python3.10/site-packages/asdf/asdf.py:589: in tree
    self._validate(asdf_object, custom=bool(tree))
../../.tox/py310/lib/python3.10/site-packages/asdf/asdf.py:619: in _validate
    schema.validate(tagged_tree, self, reading=reading)
../../.tox/py310/lib/python3.10/site-packages/asdf/schema.py:658: in validate
    validator = get_validator(schema, ctx, validators, ctx.resolver,
../../.tox/py310/lib/python3.10/site-packages/asdf/schema.py:563: in get_validator
    cls = _create_validator(validators=validators, visit_repeat_nodes=_visit_repeat_nodes)
../../.tox/py310/lib/python3.10/site-packages/asdf/schema.py:263: in _create_validator
    meta_schema = _load_schema_cached(YAML_SCHEMA_METASCHEMA_ID, extension.get_default_resolver(), False, False)
../../.tox/py310/lib/python3.10/site-packages/asdf/schema.py:464: in _load_schema_cached
    schema, url = loader(url)
../../.tox/py310/lib/python3.10/site-packages/asdf/schema.py:363: in load_schema
    resource_manager = get_config().resource_manager
../../.tox/py310/lib/python3.10/site-packages/asdf/config.py:138: in resource_manager
    self._resource_manager = ResourceManager(self.resource_mappings)
../../.tox/py310/lib/python3.10/site-packages/asdf/config.py:60: in resource_mappings
    self._resource_mappings = entry_points.get_resource_mappings()
../../.tox/py310/lib/python3.10/site-packages/asdf/entry_points.py:15: in get_resource_mappings
    return _list_entry_points(RESOURCE_MAPPINGS_GROUP, ResourceMappingProxy)
../../.tox/py310/lib/python3.10/site-packages/asdf/entry_points.py:68: in _list_entry_points
    _handle_error(e)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

e = RuntimeError('Unable to locate core schemas')

    def _handle_error(e):
>       warnings.warn(
            "{} plugin from package {}=={} failed to load:\n\n"
            "{}: {}".format(
                group,
                package_name,
                package_version,
                e.__class__.__name__,
                e,
            ),
            AsdfWarning
        )
E       asdf.exceptions.AsdfWarning: asdf.resource_mappings plugin from package asdf==2.8.1 failed to load:
E       
E       RuntimeError: Unable to locate core schemas
```

File: tox.ini [hunk 3]
```diff
@@ -96,6 +96,9 @@ deps =
     figure-!devdeps: matplotlib==3.3.2
     figure-!devdeps: mpl-animators==1.0.0
 
+    # To be fixed in a new pytest release:
+    py310: git+https://github.com/pytest-dev/pytest
```
Comment (dstansby) [inline tox.ini:100]: This doesn't seem like a great idea - why is this required?

Comment (nabobalis) [reply tox.ini:100]: This is to fix the asdf fails on the CI.

Comment (dstansby) [reply tox.ini:100]: Can we skip the ASDF tests instead, conditional on python 3.10?

Comment (nabobalis) [reply tox.ini:100]: Sure.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -85,7 +85,7 @@ stages:
 
         envs:
           - macos: py37
-          - windows: py38
+          - windows: py310
```
Comment (nabobalis) [inline azure-pipelines.yml:88]: Let us see if this is a mistake.

**Review by nabobalis — APPROVED** [review]

File: azure-pipelines.yml (hunk 4 above)
Comment (Cadair) [reply azure-pipelines.yml:88]: Isn't this likely to make it even slower?

Comment (nabobalis) [reply azure-pipelines.yml:88]: Nope.
